nmpolicy/types.go

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of the nmpolicy project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nmpolicy
+
+type PolicySpec struct {
+	Captures []CaptureSpec
+	DesiredState NMState
+}
+
+type CaptureSpec struct {
+	ID string
+	Expression string
+}
+
+type CacheState struct {
+	Captures []CaptureState
+}
+
+type GeneratedState struct {
+	Captures []CaptureState
+	DesiredState NMState
+	MetaInfo MetaInfo
+}
+
+type CaptureState struct {
+	ID string
+	State NMState
+}
+
+type NMState []byte
+
+type MetaInfo struct {
+	Version string
```

Comment:
Why version and no a timestamp?

Reply:
Version provides the info of which nmpolicy version is used to generate it, which is used by other similar tools (nftables-json).
But having a timestamp sounds useful as well, will add it too.